### PR TITLE
 feature/HW-51490-Get-user-details

### PR DESCRIPTION
**Pull request by gmeyer-hw**
HW-51490 - Get user details in the UI SDK
- Added HyperwalletUser model.
- Implemented `Hyperwallet.shared.getUser(completionHandler: (HyperwalletUser?, HyperwalletErrorType?) )` .
- Added tests coverage to `getUser(completionHandler: (HyperwalletUser?, HyperwalletErrorType?) )`
- Updated the ReadME

### CODE REVIEW

File: Sources/Model/HyperwalletUser.swift [hunk 1]
```diff
@@ -0,0 +1,847 @@
...
+    }
+
+    /// The user's role in the organization.
+    ///
+    /// - director: Director.
+    /// - owner: Owner.
+    /// - other: Other.
+    public enum BusinessContactRole: String {
+        case director = "DIRECTOR"
+        case owner = "OWNER"
```
Comment (syyang-hw) [inline Sources/Model/HyperwalletUser.swift]: ```owner``` should be after ```other```

File: Sources/Model/HyperwalletUser.swift [hunk 2]
```diff
@@ -0,0 +1,847 @@
...
+        return getField(.businessContactCountry) as? String
+    }
+
+    /// The business contact's postal code.
+    public var businessContactPostalCode: String? {
+        return getField(.businessContactPostalCode) as? String
+    }
+
+    /// The user's role in the organization
+    public var businessContactRole: BusinessContactRole? {
```
Comment (jsaini-hw) [inline Sources/Model/HyperwalletUser.swift]: Return type of this should be String too

Comment (jsaini-hw) [reply Sources/Model/HyperwalletUser.swift]: Or leave it

File: Sources/Model/HyperwalletUser.swift [hunk 3]
```diff
@@ -0,0 +1,847 @@
...
+        static func make(key: String) -> CodingKeys {
+            return CodingKeys(stringValue: key)!
+        }
+    }
+
+    /// Gets the field value
+    ///
+    /// - Parameter fieldName: The `UserField` type
+    /// - Returns: Returns the field value, or nil if none exists.
+    private func getField(_ fieldName: UserField) -> Any? {
```
Comment (jsaini-hw) [inline Sources/Model/HyperwalletUser.swift]: It should be public for if integrator just wants to use this func to getField. Also after making it public move it before Builder.

File: Tests/HyperwalletUserTests.swift [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+// The above copyright notice and this permission notice shall be included in all copies or
+// substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+// BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import Hippolyte
```
Comment (jsaini-hw) [inline Tests/HyperwalletUserTests.swift]: No copyrights for Test Classes

**Comment by gmeyer-hw** [comment]
Todo: 

- [ ] Update the Cartfile with the right `Hyperwallet-ios-sdk` target after merging the PR HW-51490

**Pull request closed without merging**